src/builders/indexable-system-page-builder.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * System page builder for the indexables.
+ *
+ * @package Yoast\YoastSEO\Builders
+ */
+
+namespace Yoast\WP\Free\Builders;
+
+use Yoast\WP\Free\Helpers\Options_Helper;
+use Yoast\WP\Free\Models\Indexable;
+
+/**
+ * Formats system pages ( search and error ) meta to indexable format.
+ */
+class Indexable_System_Page_Builder {
+
+	/**
+	 * Mapping of object type to title option keys.
+	 */
+	const OPTION_MAPPING = [
+		'search-result' => 'title-search-wpseo',
+		'404-page'      => 'title-404-wpseo',
```

Comment:
You removed the `-page` from the `search-result`. I thought this was because of the parent now being a `system-page`. But then you kept it for the `404-page` (technically `error-page`).
I'd vote for either keeping it in both or removing it in both.